src/main/java/me/cortex/jarscanner/Gui.java

```diff
@@ -45,15 +46,14 @@ private static void createAndDisplayGui() {
 
         JButton runButton = new JButton("Run!");
         runButton.addActionListener(e -> {
-            Main.run(4, searchDir, true, out -> {
-                textArea.append(out + "\n");
-                return out;
-            });
-
-            Detector.checkForStage2(s -> {
```

Comment:
Also, it looks scary to be removing the stage 2 check here, but that is already called in Main.run(), so there is no need to run it twice.